### PR TITLE
Fix dynamic category icons and update footer links

### PR DESCRIPTION
**Pull request by thebeyondr**

### CODE REVIEW

File: app/routes/_index.tsx [hunk 1]
```diff
@@ -41,24 +42,25 @@ const iconComponents: { [key: string]: LucideIcon } = {
...
 
+const DynamicCategoryIcon: React.FC<DynamicCategoryIconProps> = ({
+	category,
+	color,
+	strokeWidth,
+	size,
+}) => {
 	const CategoryIcon = iconComponents[category];
+	if (!CategoryIcon) {
+		return <AlertCircle size={14} />; // or a placeholder component
```
Comment (CJ-Rose) [inline app/routes/_index.tsx:60]: i think the `size` prop can be passed to `AlertCircle` so it doesn't throw off the alignment here
Screenshot 2024-02-07 at 1 33 27 PM

File: app/routes/_index.tsx [hunk 2]
```diff
@@ -230,40 +232,42 @@ export default function Index() {
...
+										<Progress value={report.fundedSoFar / 10} />
+										<p className="text-xs">
+											${report.totalCost - report.fundedSoFar} still needed
+										</p>
+									</CardFooter>
+								</Card>
+							</Link>
 						))}
 						{/* mapping out our 2 reports again to see how they fit */}
 						{reports.map((report: Report) => (
```
Comment (CJ-Rose) [inline app/routes/_index.tsx:273]: this extra mapping can be deleted now that we have enough HCs to fill out the container. i've removed it on my branch already but may as well get rid of it now.

**Review by CJ-Rose — CHANGES_REQUESTED** [review]

File: app/routes/_index.tsx (hunk 1 above)
Comment (thebeyondr) [reply app/routes/_index.tsx:60]: I want it to look out of place because its a sign something is wrong

File: app/components/global/footer.tsx [hunk 3]
```diff
@@ -57,15 +57,16 @@ const MobileFooter = () => {
...
+					<DrawerContent className="gap-10">
 						<DrawerHeader>
 							<div className="flex justify-center">
 								<img className="h-10 w-10" src={logo} alt="VoiceDeck Logo" />
 							</div>
 						</DrawerHeader>
 						{externalLinks.map((link) => (
-							<a
-								href={link.url}
+							<Link
```
Comment (0xRowdy) [inline app/components/global/footer.tsx:67]: I don't think you want to use the Link component for external links unless that has recently changed. Link assumes all links are internal.

We can write a component that determines whether it is internal or external though.

https://github.com/remix-run/remix/discussions/4888

Comment (thebeyondr) [reply app/components/global/footer.tsx:67]: you're right. didn't register as external to me, moving to fast!

Comment (thebeyondr) [reply app/components/global/footer.tsx]: Reverted now!

**Pull request merged**